### PR TITLE
Ethan/save navigation

### PR DESCRIPTION
**Pull request by iWatger**
Screen data persists between sessions. I ended up using user defaults because of errors beyond my knowledge with SceneStorage

### CODE REVIEW

**Comment by iWatger** [comment]
oops, fixing merge conflicts

**Comment by iWatger** [comment]
https://github.com/user-attachments/assets/80408d40-8789-4f05-85f4-24d5cb315d5c

**Comment by iWatger** [comment]
video! Apologies for the bad compression.

File: CanvasPlusPlayground/Helpers/NavigationModel.swift [hunk 1]
```diff
@@ -7,16 +7,47 @@
...
+            
+        } catch {
+            print("Error fetching selected course from user defaults \(error.localizedDescription)")
+            course = nil
+        }
+        self.selectedCourse = course
+        self.selectedCoursePage = page
+    }
+    
+    @Published var showInstallIntelligenceSheet = false
```
Comment (rahulon12) [inline CanvasPlusPlayground/Helpers/NavigationModel.swift:52]: nit: move this above the `init`

File: CanvasPlusPlayground/Model/Course.swift [hunk 2]
```diff
@@ -9,120 +9,121 @@
  https://canvas.instructure.com/doc/api/courses.html
  A course object looks like:
  {
-   // the unique identifier for the course
```
Comment (rahulon12) [inline CanvasPlusPlayground/Model/Course.swift:12]: Could you somehow remove this from your staged changes please 👀 😿

**Review by rahulon12 — CHANGES_REQUESTED** [review]

File: CanvasPlusPlayground/Model/Course.swift (hunk 2 above)
Comment (iWatger) [reply CanvasPlusPlayground/Model/Course.swift]: fixed!

File: CanvasPlusPlayground/Helpers/NavigationModel.swift (hunk 1 above)
Comment (iWatger) [reply CanvasPlusPlayground/Helpers/NavigationModel.swift]: fixed!

File: CanvasPlusPlayground/Views/CourseListView.swift [hunk 3]
```diff
@@ -53,6 +58,9 @@ struct CourseListView: View {
             }
             .interactiveDismissDisabled()
         }
+
```
Comment (rahulon12) [inline CanvasPlusPlayground/Views/CourseListView.swift:61]: nit: 
```suggestion
```

**Review by rahulon12 — APPROVED** [review]

**Pull request merged**